test/unit/media.js

```diff
@@ -229,3 +229,26 @@ test('should add the source hanlder interface to a tech', function(){
   tech.dispose();
   ok(disposeCalled, 'the handler dispose method was called when the tech was disposed');
 });
+
+test('should handle unsupported sources with the source hanlder API', function(){
+  var mockPlayer = {
+    off: this.noop,
+    trigger: this.noop
+  };
+
+  // Define a new tech class
+  var Tech = videojs.MediaTechController.extend();
+  // Extend Tech with source handlers
+  vjs.MediaTechController.withSourceHandlers(Tech);
+
+  Tech.nativeSourceHandler = {
+    handleSource: function(){ usedNative = true; }
```

Comment:
I don't love the weird indirection here. Why not just do the assertion right in this callback and set the number of expected assertions to 1?